### DISCUSSION TITLE
How to configure MQTT bridge in version 5.0.0 Docker with environmental variables

### DISCUSSION THREAD

**Discussion by matthewGEA** [emqx/emqx#8338] [Q&A] [answered]
I'm using this as a reference:
https://github.com/emqx/emqx-docs/blob/release-5.0/en_US/admin/cfg.md#connector-mqtt-connector

I am using a docker compose file to create the desired configuration.

1) I do not see an option to specify CLIENTID.  Was this removed in version 5?  Before, it was:
 - EMQX_BRIDGE__MQTT__$name__CLIENTID

2) How can we specify a ${field} syntax variable when docker compose will try to resolve that variable before passing the value to emqx?  Examples include REMOTE_TOPIC="${topic} or PAYLOAD="${payload}"

Below are my environmental variables for an egress bridge

      - EMQX_BRIDGES__MQTT__REMOTESVR__DIRECTION=egress
      - EMQX_BRIDGES__MQTT__REMOTESVR__ENABLE=true
      - EMQX_BRIDGES__MQTT__REMOTESVR__LOCAL_TOPIC="localtopic/#"
      - EMQX_BRIDGES__MQTT__REMOTESVR__REMOTE_TOPIC="${topic}"
      - EMQX_BRIDGES__MQTT__REMOTESVR__REMOTE_QOS=0
      - EMQX_BRIDGES__MQTT__REMOTESVR__RETAIN=false
      - EMQX_BRIDGES__MQTT__REMOTESVR__PAYLOAD="${payload}"
      - EMQX_BRIDGES__MQTT__REMOTESVR__CONNECTOR__MODE=cluster_shareload
      - EMQX_BRIDGES__MQTT__REMOTESVR__CONNECTOR__SERVER="${UPSTREAMSERVERADDR}:1883"
      - EMQX_BRIDGES__MQTT__REMOTESVR__CONNECTOR__PROTO_VER=v3
      - EMQX_BRIDGES__MQTT__REMOTESVR__CONNECTOR__BRIDGE_MODE=true
      - EMQX_BRIDGES__MQTT__REMOTESVR__CONNECTOR__USERNAME=tester
      - EMQX_BRIDGES__MQTT__REMOTESVR__CONNECTOR__PASSWORD=password
      - EMQX_BRIDGES__MQTT__REMOTESVR__CONNECTOR__CLEAN_START=false

**Comment by thalesmg** [comment, marked as answer]
Hi @matthewGEA !

You probably want to escape the dollar sign by using `$${topic}`.

Reference: https://docs.docker.com/compose/compose-file/compose-file-v2/#variable-substitution

**Reply by matthewGEA** [reply]
Yes, doing $${topic} did the trick.

Do you know how to configure the CLIENT_ID for the bridge?

**Reply by thalesmg** [reply]
It seems like in EMQX 5.0 it's not possible to directly define the clientid.  It'll be computed from the internal ID of the bridge and the node name.

For instance, for a MQTT bridge called `my_egress_mqtt_bridge` running in a node `emqx@127.0.0.1`, this could take the form of `bridge:mqtt:my_egress_mqtt_bridge:1:emqx@127.0.0.1`, as an example.
